lmp/jobserv.yml

```diff
@@ -615,6 +615,7 @@ params:
   DISTRO: lmp
   SOTA_CLIENT: aktualizr-lite
   OSTREE_BRANCHNAME: lmp
+  SSTATE_CACHE_MIRROR: https://storage.googleapis.com/lmp-cache
```

Comment:
Serve the public LmP by version is something that will not scale well, this will not work transparently for the end user without change this file https://github.com/foundriesio/lmp-manifest/pull/255/files for each LmP release.
My suggestion is merging all the sstate folders with overlayfs and made this available on the http server.

Reply:
i don't control the implementation of `https://storage.googleapis.com/lmp-cache` so i can't do an overlay for that. And we've learned that we can't control the size of sstate cache w/o versioning it. It will simply grow unbounded over time.

Reply:
I understand and agree with versioning the cache  but I am thinking in a clean way of sharing the cache.  Looking again on that and I can add some logic on the https://github.com/foundriesio/lmp-manifest/pull/255 to handle the version without user intervention so we can avoid the overlay for now.

Reply:
A similar approach can be done in setup-environment when generating auto.conf or similar, to use the correct path.

Reply:
yeah have already add it

Reply:
I think the problem is that our CI scripts will default this to what we want for customers - https://github.com/foundriesio/ci-scripts/blob/master/lmp/bb-config.sh#L20 - so we have to *overrride* that for CI.

Reply:
We can use one or the other in the CI so we can rewrite the https://github.com/foundriesio/ci-scripts/pull/239/files#diff-f12b382648c4eb1512e3828fbc501e21e138c7b08fba2f6d74410c22ae84bf4aR207-R216

```
if [[ "$SSTATE_CACHE_MIRROR" == "https://"* ]]  ; then
	cat << EOFEOF >> conf/local.conf
SSTATE_MIRRORS = "file://.* ${SSTATE_CACHE_MIRROR}/v$LMP_VER-sstate-cache/PATH"
EOFEOF
elif [ -d $SSTATE_CACHE_MIRROR ]; then
	cat << EOFEOF >> conf/local.conf
SSTATE_MIRRORS = "file://.* file://${SSTATE_CACHE_MIRROR}/PATH"
EOFEOF
fi
```
and this will prioritize the http mirror over the nfs

Reply:
yes, but the problem will still remain that unless I set it to https://.... here, it will get defaulted to a directory above with:
```
SSTATE_CACHE_MIRROR="${SSTATE_CACHE_MIRROR-/sstate-cache-mirror/v$LMP_VER-sstate-cache}"
```
and in the case of CI - its probably going to be a one or the other.

Reply:
another approach, is taking the value defined by the setup-environment-internal https://github.com/foundriesio/lmp-manifest/pull/255, it will adds the http mirros as a weak default so it will be used for all build if no other configuration exist.
In this approach we need to setup the nfs mirror (to the directory) for the non-public lmp builds on the ci.

Reply:
I think I'm missing something or thinking about this wrong. Maybe you can help show me in this PR how this can be simplified. The problem I have is that every single customer uses this script and doesn't not have `SSTATE_CACHE_MIRROR` defined. So I *think* I need to default this to what they all currently need `/sstate-cache-mirror/v$LMP_VER-sstate-cache`. However, for this public LmP CI job we need this to not be via HTTP. 

Also - as I commented in your PR - the the value of LMP_VER for this and that should be two different things?